### DISCUSSION TITLE
Does Zeek have a util to parse serialized/stringified JSON?

### DISCUSSION THREAD

**Discussion by brsolomon-deloitte** [zeek/zeek#1944] [How-Tos]
We are using signature matching to write a custom log file that includes the `request_body` and `reply_body` for each request/response pair.

Very briefly this ultimately looks like:

```
event signature_match(state: signature_state, msg: string, data: string)
```

Where `data` points to the HTTP POST body or the reply content, depending on state.

Those two attributes are currently `string`, e.g.

```
    type Info: record {
        sig_id:         string      &log;
        uid:            string      &log;
        ts:             time        &log;
        id:             conn_id     &log;
        uri:            string      &log &optional;
        request_body:   string      &log;
        reply_body:     string      &log &optional;
    };
```

with the resulting log looking like (after jq'd):

```
{
  "sig_id": "json-rpc-request-http",
  "uid": "CGRPeG247M3GHfVcQk",
  "ts": 1642712771.418473,
  "id.orig_h": "127.0.0.1",
  "id.orig_p": 49510,
  "id.resp_h": "127.0.0.1",
  "id.resp_p": 7777,
  "uri": "/",
  "request_body": "{\"jsonrpc\": \"2.0\", \"method\": \"subtract\", \"params\": [23, 42], \"id\": 2}",
  "reply_body": "{\"id\": 2, \"jsonrpc\": \"2.0\", \"result\": -19}"
}
```

This is OK, but `request_body` and `reply_body` are serialized/escaped JSON _strings_. It would be great if we could parse that JSON from `data` and write it out as a bona fide JSON object, as the value for the `request_body` and `reply_body` keys.

In Zeek utils I see `to_json` but no `from_json`. (nor do I really have a good idea of what the `type` would be. we are just interested in following the JSON-RPC standard, so assume the content could be any free-form JSON object or array.

Does Zeek offer a means to do something like this?

**Comment by 0xxon** [comment]
No, Zeek currently has no built-in way of parsing json. This is definitely something that we should fix (and something that has come up in the past).
